src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
+        for (final Talk talk : talks.active()) {
+            final List<String> requests = talk.read().xpath("//request");
+            if (requests.isEmpty()) {
+                int index = 0;
+                final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+                if (logs.isEmpty()) {
+                    index = maxTalk + 1;
+                } else {
+                    index = this.getMaxLogIndex(logs) + 1;
+                }
+                talk.modify(
+                    new Directives().xpath("//talk").add("request")
+                        .attr(INDEX, Integer.toString(index))
+                        .attr("id", this.createRequestId())
+                        .add("type").set(INDEX)
+                        .up()
+                        .add("args")
+                );
+                maxTalk += 1;
+            }
+        }
+    }
+
+    /**
+     * Returns the greatest index of all log children of an archive node.
+     * @param logs A list of XML objects for individual log nodes (e. g.
+     *  <log id="3" index="1" title="title3"/>
+     * @return Highest value of the index attribute of all log nodes contained
+     *  in the logs list.
+     */
+    private int getMaxLogIndex(final List<XML> logs) {
```

Comment:
I don't really like/understand current design, with so many `max*` methods. I would create just two: `max(Talks)` and `max(Talk)`. The first one is going through them one by one. The second one I would design in a functional style (using `Ints` from Guava):

```
return Ints.max(
  talk.read().xpath("/talk/archive/log/@index|/talk/request/@index"),
  new Function<String, Integer>() {
    public Integer transform(String input) {
      return Integer.parseInt(input);
    }
  }
);
```